packages/components/nodes/vectorstores/Postgres/driver/PGVector.ts

```diff
@@ -0,0 +1,107 @@
+import { VectorStoreDriver } from './Base'
+import { getCredentialData, getCredentialParam } from '../../../../src'
+import { PGVectorStore, PGVectorStoreArgs } from '@langchain/community/vectorstores/pgvector'
+import { Document } from '@langchain/core/documents'
+import { PoolConfig } from 'pg'
+
+export class PGVectorDriver extends VectorStoreDriver {
+    static CONTENT_COLUMN_NAME_DEFAULT: string = 'pageContent'
+
+    protected _postgresConnectionOptions: PoolConfig
+
+    protected async getPostgresConnectionOptions() {
+        if (!this._postgresConnectionOptions) {
+            const credentialData = await getCredentialData(this.nodeData.credential ?? '', this.options)
+            const user = getCredentialParam('user', credentialData, this.nodeData)
+            const password = getCredentialParam('password', credentialData, this.nodeData)
+            const additionalConfig = this.nodeData.inputs?.additionalConfig as string
+
+            let additionalConfiguration = {}
+
+            if (additionalConfig) {
+                try {
+                    additionalConfiguration = typeof additionalConfig === 'object' ? additionalConfig : JSON.parse(additionalConfig)
+                } catch (exception) {
+                    throw new Error('Invalid JSON in the Additional Configuration: ' + exception)
+                }
+            }
+
+            this._postgresConnectionOptions = {
+                ...additionalConfiguration,
+                host: this.nodeData.inputs?.host as string,
+                port: this.nodeData.inputs?.port as number,
+                user: user,
+                password: password,
+                database: this.nodeData.inputs?.database as string
+            }
+        }
+
+        return this._postgresConnectionOptions
+    }
+
+    async getArgs(): Promise<PGVectorStoreArgs> {
```

Comment:
for `pgvector`, can we allow the option for choosing [distance strategy](https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-community/src/vectorstores/pgvector.ts#L40)? https://github.com/FlowiseAI/Flowise/issues/2198

Reply:
probably, i don't know about it but it's a good learning opportunity ;)

Reply:
@HenryHengZJ done in https://github.com/FlowiseAI/Flowise/pull/3367/commits/967f0bb53db4c514ed1d3b2756c431bf35257fd3

⚠️ not tested ⚠️

Reply:
thanks! will test it out